pkg/api/v20160930/validate.go

```diff
@@ -48,6 +48,11 @@ func (a *AgentPoolProfile) Validate() error {
 	if e := validateName(a.VMSize, "AgentPoolProfile.VMSize"); e != nil {
 		return e
 	}
+	if orchestratorType == Kubernetes {
```

Comment:
I am working on a solution that doesn't need to change string comparison. So feel free to check in.